homeassistant/components/apcupsd/config_flow.py

```diff
@@ -0,0 +1,161 @@
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.config_entries import (
+    CONN_CLASS_LOCAL_POLL,
+    SOURCE_IMPORT,
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_RESOURCES
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from . import DOMAIN, APCUPSdData
+from .sensor import SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """APCUPSd integration config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # Test if connection to the host is ok and get the current status for later
+            # configuration.
+            data_service = APCUPSdData(user_input[CONF_HOST], user_input[CONF_PORT])
+            try:
+                status: dict[str, Any] | None = await self.hass.async_add_executor_job(
+                    lambda: data_service.status
+                )
+                if status is None:
+                    return self.async_abort(reason="no_status")
+
+                # We _try_ to use the serial number of the UPS as the unique id since
+                # this field is not guaranteed to exist on all APC UPS models.
+                if "SERIALNO" in status:
+                    await self.async_set_unique_id(status["SERIALNO"])
+                    self._abort_if_unique_id_configured()
+
+                # If we import from YAML, user_input may contain CONF_RESOURCES which
+                # should be set as options.
+                # Note that it may contain unavailable resources, so we need to filter
+                # them out and log warnings for them.
+                options = None
+                if self.source == SOURCE_IMPORT and CONF_RESOURCES in user_input:
+                    available_resources = set(data_service.status.keys())
+                    resources = []
+                    for resource in user_input[CONF_RESOURCES]:
+                        if resource not in available_resources:
+                            _LOGGER.warning(
+                                "Sensor type: %s does not appear in the APCUPSd status output and will be ignored",
+                                resource,
+                            )
+                            continue
+                        resources.append(resource)
+
+                    options = {CONF_RESOURCES: resources}
```

Comment:
I'd update the item in the config entry data instead, since we won't be adding an options flow.

Reply:
Yes, merged this into config entry data.